include/pmacc/meta/ExecuteIf.hpp

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2013-2021 Rene Widera, Benjamin Worpitz, Alexander Grund
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <utility>
+
+namespace pmacc
+{
+    namespace meta
+    {
+        namespace detail
+        {
+            /** Conditional functor execution
+             *
+             * @tparam T_condition condition for executing a functor
+             */
+            template<bool T_condition>
+            struct IfElse;
+
+            /** Specialization if condition is true
+             *
+             * Execute trueFunctor passed to operator()
+             */
+            template<>
+            struct IfElse<true>
+            {
+                template<typename T_ConditionTrueFunctor, typename T_ConditionFalseFunctor, typename... T_Args>
+                void operator()(T_ConditionTrueFunctor&& trueFunctor, T_ConditionFalseFunctor&&, T_Args&&... args)
```

Comment:
This should be `HDINLINE` and the other specialization as well.

Reply:
Also offline discussion. How does it work at all for CUDA? If interpreted as written, this function can't be called on device at all. So does the compiler inline more now? Perhaps related to https://github.com/alpaka-group/alpaka/issues/1408#issuecomment-914880228

Reply:
done